src/win/fs.c

```diff
@@ -142,6 +142,57 @@ void uv_fs_init(void) {
 }
 
 
+static unsigned int fs__decode_wtf8_char(const char** input) {
+  unsigned int code_point;
+  unsigned char b = **input, b2, b3, b4;
+  if (b >= 0x00 && b <= 0x7F) {
+    code_point = b;
+  } else if (b >= 0xC2 && b <= 0xDF) {
+    assert((*input)[1]);
+    b2 = *++*input;
+    code_point = ((b & 0x1F) << 6) + (b2 & 0x3F);
+  } else if (b >= 0xE0 && b <= 0xEF) {
+    assert((*input)[1] && (*input)[2]);
+    b2 = *++*input;
+    b3 = *++*input;
+    code_point = ((b & 0x0F) << 12) + ((b2 & 0x3F) << 6) + (b3 & 0x3F);
+  } else if (b >= 0xF0 && b <= 0xF0) {
+    assert((*input)[1] && (*input)[2] && (*input)[3]);
+    b2 = *++*input;
+    b3 = *++*input;
+    b4 = *++*input;
+    code_point = ((b & 0x07) << 18) + ((b2 & 0x3F) << 12) +
+      ((b3 & 0x3F) << 6) + (b4 & 0x3F);
+  }
+  assert(code_point <= 1114111);
```

Comment:
```suggestion
  assert(code_point <= 1114111); /* 0x10ffff */
```
(though, again, we should make this return an error, not assert it)